src/abbrev.rs

```diff
@@ -683,34 +648,17 @@ mod tests {
     #[test]
     fn test_parse_null_attribute_specification_ok() {
         let buf = [0x00, 0x00, 0x01];
-        let (rest, _) = AttributeSpecification::parse_null(&buf)
+        let (rest, attr) = AttributeSpecification::parse(&buf)
             .expect("Should parse null attribute specification");
+        assert!(attr.is_none());
         assert_eq!(rest, [0x01]);
     }
 
     #[test]
-    fn test_parse_null_attribute_specification_form_nonzero() {
-        let buf = [0x00, 0x01];
-        match AttributeSpecification::parse_null(&buf) {
-            Err(Error::ExpectedZero) => {}
-            otherwise => panic!("Unexpected result: {:?}", otherwise),
-        };
-    }
-
-    #[test]
-    fn test_parse_null_attribute_specification_name_nonzero() {
-        let buf = [0x01, 0x00];
-        match AttributeSpecification::parse_null(&buf) {
-            Err(Error::ExpectedZero) => {}
-            otherwise => panic!("Unexpected result: {:?}", otherwise),
-        };
-    }
-
-    #[test]
     fn test_parse_attribute_specifications_name_zero() {
         let buf = [0x00, 0x01, 0x00, 0x00];
         match AttributeSpecification::parse(&buf) {
-            Err(Error::AttributeNameZero) => {}
+            Err(Error::ExpectedZero) => {}
```

Comment:
Not sure about this change since it makes the handling for zeroes in name/form asymmetrical. As it is, the AttributeNameZero enum value is now unused and could be deleted.